examples/AffinePlaneCurve.jl

```diff
@@ -0,0 +1,124 @@
+module AffinePlaneCurveModule
+
+using Oscar, Markdown
+
+export AffinePlaneCurve, AffinePlaneCurve_degree, AffinePlaneCurve_Jacobian, AffinePlaneCurve_tangent, AffinePlaneCurve_issmoothpoint, AffinePlaneCurve_union
+
+####################################################
+#
+#We follow the lecture notes by Andreas Gathmann: https://www.mathematik.uni-kl.de/~gathmann/en/curves.php
+#
+####################################################
+mutable struct AffinePlaneCurve{S}
+  eq::Oscar.MPolyElem                      #Equation of the curve (polynomial in two variables)
+  degree::Union{String, Int64}             #degree of the equation of the curve
+  dimension::Int64                         #dimension of the curve (as a variety)
+  function AffinePlaneCurve(eq::Oscar.MPolyElem{T}) where {T <: FieldElem}
+    r = new{T}()
+    r.eq = eq
+    if nvars(parent(r.eq))!=2              #If the ambiant ring does not have two variables, show an error.
+       error("The defining equation must belong to a ring with two variables")
+    else
+       r.degree = "not computed yet"
+       r.dimension = 1                    #since C is a plane curve, the dimension is always 1
+       return r
+    end
+  end
+end
+
+function Base.show(io::IO, C::AffinePlaneCurve)
+  println(io, "Affine plane curve defined by ", C.eq)
+end
+
+######################################################
+#Compute the degree of the equation of the curve if not already known, and show it.
+
+
+@doc Markdown.doc"""
+   AffinePlaneCurve_degree(C::AffinePlaneCurve)
+   > Given an affine plane curve C, returns the degree of the defining polynomial.
+"""
+function AffinePlaneCurve_degree(C::AffinePlaneCurve)
+  if typeof(C.degree)==Int64
+     println(C.degree)
+  else
+  C.degree=total_degree(C.eq)
+  println(C.degree)
+  end
+end
+
+######################################################
+#Compute the Jacobian Ideal of C
+
+@doc Markdown.doc"""
+   AffinePlaneCurve_Jacobian(C::AffinePlaneCurve)
+   > Given an affine plane curve C, returns the Jacobian ideal of the defining polynomial.
+"""
+function AffinePlaneCurve_Jacobian(C::AffinePlaneCurve)
+ return jacobi_ideal(C.eq)
+end
+
+######################################################
+#To check if a point is smooth: return true if the point is a smooth point on the curve C, and false otherwise.
+
+@doc Markdown.doc"""
+   AffinePlaneCurve_Jacobian(C::AffinePlaneCurve, P::Array{S,1}) where S<: FieldElem
+   > Given an affine plane curve C and a point P, returns false if the point is not in the zero locus of the defining equation or is a singular point of C, and true if it is a smooth point of the curve.
+"""
+function AffinePlaneCurve_issmoothpoint(C::AffinePlaneCurve{S}, P::Array{S,1}) where S<: FieldElem
```

Comment:
I think it may make sense to define a type for a point on a plane curve, rather than to work with an anonymous array. This would allow you to define whatever arithmetic you actually want on points, rather than be stuck with whatever Julia defines.